src/gsSolver/gsSumOfOperatorsOp.h

```diff
@@ -0,0 +1,129 @@
+///
+/// @ingroup Solver
+template<typename T>
+class gsSumOfOperatorsOp GISMO_FINAL : public gsLinearOperator<T>
+{
+    typedef typename gsLinearOperator<T>::Ptr BasePtr;
+public:
+
+    /// Shared pointer for gsSumOfOperatorsOp
+    typedef memory::shared_ptr<gsSumOfOperatorsOp> Ptr;
+
+    /// Unique pointer for gsSumOfOperatorsOp
+    typedef memory::unique_ptr<gsSumOfOperatorsOp> uPtr;
+
+    /// Empty constructor. To be filled with addOperator()
+    gsSumOfOperatorsOp() : m_sz(0), m_ops(0) {}
+
+    /// Constructor taking a vector of Linear Operators
+    gsSumOfOperatorsOp(const std::vector<BasePtr>& ops)
+        : m_sz(ops.size()), m_ops(ops)
+    {
+#ifndef NDEBUG
+        for (index_t i=0; i<m_sz; ++i)
+        {
+            GISMO_ASSERT ( m_ops[0]->rows() == m_ops[i]->rows() && m_ops[0]->cols() == m_ops[i]->cols(),
+                           "Dimensions of the operators do not fit." );
+        }
+#endif
+    }
+
+    /// Constructor taking two Linear Operators
+    gsSumOfOperatorsOp(const BasePtr& op0, const BasePtr& op1)
+        : m_sz(2), m_ops(2)
+    {
+        GISMO_ASSERT ( op0->rows() == op1->rows() && op0->cols() == op1->cols(), "Dimensions of the operators do not fit." );
+        m_ops[0] = op0; m_ops[1] = op1;
+    }
+
+    /// Constructor taking three Linear Operators
+    gsSumOfOperatorsOp(const BasePtr& op0, const BasePtr& op1, const BasePtr& op2 )
+        : m_sz(3), m_ops(3)
+    {
+        GISMO_ASSERT ( op0->rows() == op1->rows() && op0->cols() == op1->cols()
+                        && op0->rows() == op2->rows() && op0->cols() == op2->cols(), "Dimensions of the operators do not fit." );
+        m_ops[0] = op0; m_ops[1] = op1; m_ops[2] = op2;
+    }
+
+    /// Make command returning a smart pointer
+    static uPtr make()
+    { return memory::make_unique( new gsSumOfOperatorsOp() ); }
+
+    /// Make command returning a smart pointer
+    static uPtr make( const std::vector<BasePtr>& ops )
+    { return memory::make_unique( new gsSumOfOperatorsOp( ops ) ); }
+
+    /// Make command returning a smart pointer
+    static uPtr make( const BasePtr& op0, const BasePtr& op1 )
+    { return memory::make_unique( new gsSumOfOperatorsOp( op0, op1 ) ); }
+
+    /// Make command returning a smart pointer
+    static uPtr make( const BasePtr& op0, const BasePtr& op1, const BasePtr& op2 )
+    { return memory::make_unique( new gsSumOfOperatorsOp( op0, op1, op2 ) ); }
+
+    /// Add another operator
+    void addOperator( const BasePtr& op )
+    {
+        GISMO_ASSERT ( m_sz == 0 || ( op->rows() == m_ops[0]->rows() && op->cols() == m_ops[0]->cols() ),
+                       "Dimensions of the operators do not fit." );
+        ++m_sz;
+        m_ops.push_back( op );
+    }
+
+    void apply(const gsMatrix<T> & input, gsMatrix<T> & result) const
+    {
+        GISMO_ASSERT ( m_sz>0, "gsSumOfOperatorsOp::apply does not work for 0 operators." );
+
+        // Here, we could make a permanently allocated vector
```

Comment:
(general comment)
This would happen if the function was returning "result" instead of void..